includes/polling/applications/mysql.inc.php

```diff
@@ -99,6 +99,9 @@
 ];
 
 $data = explode("\n", $mysql);
+if (sizeof($data) > 0 && strpos($data[0], 'ERROR') === 0) {
```

Comment:
Probably the correct place to update is https://github.com/librenms/librenms/blob/master/includes/polling/functions.inc.php#L447

and change it to check if the string starts with the error message and then mark it as ERROR.  That way we don't delete the additional context after the ERROR message.

The other option would be to set $state directly to the update_application() call.

Reply:
From how I understand that function there is no real way to save an error-message in the `application` table. Also the `app_state` would only hold 32 chars and the `app_state` 8... So that'd not fit with a full error-message.

We could modify the log_event function that is later called to include the error-message from the response, but i'm not sure what implication that would have for all the other apps really.

If I move it to the $status parameter of the update_application function it would not trigger the alerts (I think) since there is no comparison here https://github.com/librenms/librenms/blob/master/includes/polling/functions.inc.php#L474 that references $status. Should I add that so it also triggers when I update the $status?

Reply:
I'm saying inside update_applicatuon(), status is detected.  You need to improve the detection to see if it starts with one of the listed strings set the status to the exact string.  The full message should be in the response field.

Hope that is clear enough, I'm on my phone.